src/dos/program_more.cpp

```diff
@@ -60,15 +61,79 @@ bool MORE::ParseCommandLine(MoreOutputFiles &output)
 	cmd->FillVector(params);
 
 	// Check if specified tabulation size
-	if (!params.empty()) {
-		const auto &param = params[0];
-		if ((starts_with("/t", param) || starts_with("/T", param)) &&
-		    (param.length() == 3) && (param.back() >= '1') &&
-		    (param.back() <= '9')) {
-			// FreeDOS extension - custom TAB size
-			output.SetTabSize(static_cast<uint8_t>(param.back() - '0'));
+	while (!params.empty()) {
+		auto is_option_extended = [](const std::string& param) {
+			return param == "/e" || param == "/E";
+		};
+		auto is_option_clear = [](const std::string& param) {
+			return param == "/c" || param == "/C";
+		};
+		auto is_option_expand_form_feed = [](const std::string& param) {
+			return param == "/p" || param == "/P";
+		};
+		auto is_option_squish = [](const std::string& param) {
+			return param == "/s" || param == "/S";
+		};
+		auto is_option_start_line = [](const std::string& param) {
+			if ((param.length() < 2) || (param[0] != '+')) {
+				return false;
+			}
+			for (size_t i = 1; i < param.length(); ++i) {
+				if (!std::isdigit(param[i])) {
+					return false;
+				}
+			}
+			return true;
+		};
+		auto is_option_tab_size = [](const std::string& param) {
+			return (param.length() == 3) &&
+			       (starts_with("/t", param) || starts_with("/T", param)) &&
+			       (param.back() >= '1') && (param.back() <= '9');
+		};
+
+		const auto& param = params[0];
+		if (is_option_extended(param)) {
+			output.SetOptionExtendedMode(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_clear(param)) {
+			output.SetOptionClear(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_expand_form_feed(param)) {
+			output.SetOptionExpandFormFeed(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_squish(param)) {
+			output.SetOptionSquish(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_start_line(param)) {
+			int32_t line_num = 0;
+			try {
+				line_num = std::stoi(param);
+			} catch (...) {
+				result_errorcode = DOSERR_FUNCTION_NUMBER_INVALID;
+				WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"),
+				         param.c_str());
+				return false;
+			}
+			output.SetOptionStartLine(static_cast<uint32_t>(line_num));
```

Comment:
The standard library finally gives us a safe `std::string` to integer parser, but then forces us to jam exception handling boiler plate around it, which completely destroys readability.

Would you be able to put this into a `string_util.h` helper?

I'm thinking the interface should return a `std::optional<int>` (`int`, because that's the type-size returned by `std::stoi`).

It would let us use it something like this:

```c++
if (const auto p = to_int(param); p) {
    const auto line_num = check_cast<uint32_t>(*p);
    output.SetOptionStartLine(line_num);
} else {
    // error messages and codes 
}
```

Reply:
OK, changed.